### PR TITLE
[typescript-axios] Add stringEnums option

### LINKED ISSUE #10943: [REQ] [typescript-axios] Object value enums
### Is your feature request related to a problem? Please describe.

While the typescript interfaces generated for the model can be fully compatible with client side models, this is not the case with enums as different string enums cannot be compatible on the client side. Typescript enums are problematic, and the typescript community has a better solution.

Implemented this would allow the client side models and server side models to be entirely decoupled, as enums are the only thing standing in the way of that at the moment.

I realize that many people likely use the server models directly, but a flexible approach would be preferable.

## Describe the solution you'd like

The typescript-angular generator actually already implements this by default (stringEnums = false). It creates an object value enum as such:

```
export interface Model {
  type: Model.TypeEnum;
}
export namespace Model{
  export type TypeEnum = 'ONE' | 'TWO' | 'THREE';
  export const TypeEnum = {
    ONE: 'ONE' as TypeEnum,
    ONE: 'TWO' as TypeEnum,
    ONE: 'THREE' as TypeEnum,
  } as const;
}
```

## Alternative Solutions

It would be even greater if this could be extended to all typescript generators, but I'm not sure if their functionality is centralized in any way.

I did take a quick look at the source, but my Java-fu is weak and I can't find the obvious place to implement the change.

### PR DESCRIPTION
**Pull request by ty-v1**
This PR adds `stringEnums` option to typescript-axios generator.
Closes #10943

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java [hunk 1]
```diff
@@ -127,6 +131,15 @@ public void processOpts() {
             }
         }
 
+        if (additionalProperties.containsKey(STRING_ENUMS)) {
+            this.stringEnums = Boolean.parseBoolean(additionalProperties.get(STRING_ENUMS).toString());
+            additionalProperties.put("stringEnums", this.stringEnums);
+            if (this.stringEnums) {
+                enumSuffix = "";
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:138]: Why is this needed?

File: samples/client/petstore/typescript-axios/builds/composed-schemas/api.ts [hunk 2]
```diff
@@ -79,16 +79,14 @@ export interface Dog {
...
-    * @enum {string}
-    */
-export enum DogBreedEnum {
-    Dingo = 'Dingo',
-    Husky = 'Husky',
-    Retriever = 'Retriever',
-    Shepherd = 'Shepherd'
-}
+export const DogBreedEnum = {
+    Dingo: 'Dingo',
```
Comment (macjohnny) [inline samples/client/petstore/typescript-axios/builds/composed-schemas/api.ts:83]: Please also add samples with stringEnums=true

**Review by macjohnny — COMMENTED** [review]
Thanks for your contribution!

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java (hunk 1 above)
Comment (ty-v1) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:138]: There is no particular reason.
I reused code from [here](https://github.com/OpenAPITools/openapi-generator/blob/b7c8de94250905c24b4b386a1054a3b5e8bb6537/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNestjsClientCodegen.java#L145-L152).

Comment (ty-v1) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:138]: I’ll remove this code for the following reasons:
- Users may be confused if the name of the enum changes depending on the options.
- In this generator, `classEnumSeparator` is not used.

File: modules/openapi-generator/src/main/resources/typescript-axios/modelEnum.mustache [hunk 3]
```diff
@@ -7,17 +7,5 @@
...
-    {{#enumDescription}}
-    /**
-    * {{.}}
-    */
-    {{/enumDescription}}
-    {{{name}}} = {{{value}}}{{^-last}},{{/-last}}
-{{/enumVars}}
-{{/allowableValues}}
-}
+{{^isBoolean}}{{^stringEnums}}{{>modelObjectEnum}}{{/stringEnums}}{{#stringEnums}}{{>modelStringEnum}}{{/stringEnums}}
```
[the remark this answers is not in the document]
Comment (ty-v1) [reply modules/openapi-generator/src/main/resources/typescript-axios/modelEnum.mustache]: Thank you. 
I do not know much about Mustache.js, so this is very helpful.

[the remark this answers is not in the document]
Comment (ty-v1) [reply modules/openapi-generator/src/main/resources/typescript-axios/modelEnum.mustache]: I fixed it with d83c036263b9dc1600caa8ece47938ce90bcb9eb

File: samples/client/petstore/typescript-axios/builds/composed-schemas/api.ts (hunk 2 above)
Comment (ty-v1) [reply samples/client/petstore/typescript-axios/builds/composed-schemas/api.ts:83]: Sorry, I forget to add samples.

**Comment by macjohnny** [comment]
@ty-v1 can you update the samples please?

**Comment by ty-v1** [comment]
I could not work on it because I bought a new PC.
Now that the setup is complete, I'll update the samples.

File: samples/client/petstore/typescript-axios/builds/composed-schemas/api.ts (hunk 2 above)
Comment (ty-v1) [reply samples/client/petstore/typescript-axios/builds/composed-schemas/api.ts:83]: I added samples with the name `with-string-enums`.

**Review by macjohnny — APPROVED** [review]
LGTM

File: modules/openapi-generator/src/main/resources/typescript-axios/modelObjectEnum.mustache [hunk 4]
```diff
@@ -1,14 +1,14 @@
-export const {{classname}}{{enumName}} = {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-axios/modelObjectEnum.mustache]: does this work when there are multiple enums defined in a model?

Comment (ty-v1) [reply modules/openapi-generator/src/main/resources/typescript-axios/modelObjectEnum.mustache]: Yes. 
I generated codes using the schema below, and correct codes are generated.

### openapi schema
```yaml
openapi: 3.0.3
info:
  title: Enum
  description: Enum
  version: 1.0.0
paths:
  /items:
    get:
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Response'
  /generic:
    get:
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  m1:
                    type: string
                    enum: [ a, b, c ]
                  m2:
                    type: number
                    enum: [ 1, 2, 3 ]
components:
  schemas:
    Response:
      type: object
      properties:
        enum1:
          type: string
          enum: [ a, b, c ]
        enum2:
          type: string
          enum: [ e, d, f ]
```

### `stringEnums = false`
```typescript
export interface InlineResponse200 {
    'm1'?: InlineResponse200M1Enum;
    'm2'?: InlineResponse200M2Enum;
}

export const InlineResponse200M1Enum = {
    A: 'a',
    B: 'b',
    C: 'c'
} as const;

export type InlineResponse200M1Enum = typeof InlineResponse200M1Enum[keyof typeof InlineResponse200M1Enum];
export const InlineResponse200M2Enum = {
    NUMBER_1: 1,
    NUMBER_2: 2,
    NUMBER_3: 3
} as const;

export type InlineResponse200M2Enum = typeof InlineResponse200M2Enum[keyof typeof InlineResponse200M2Enum];

export interface Response {
    'enum1'?: ResponseEnum1Enum;
    'enum2'?: ResponseEnum2Enum;
}

export const ResponseEnum1Enum = {
    A: 'a',
    B: 'b',
    C: 'c'
} as const;

export type ResponseEnum1Enum = typeof ResponseEnum1Enum[keyof typeof ResponseEnum1Enum];
export const ResponseEnum2Enum = {
    E: 'e',
    D: 'd',
    F: 'f'
} as const;

export type ResponseEnum2Enum = typeof ResponseEnum2Enum[keyof typeof ResponseEnum2Enum];
```


### `stringEnums = true`

```typescript
export interface InlineResponse200 {
    'm1'?: InlineResponse200M1Enum;
    'm2'?: InlineResponse200M2Enum;
}

export enum InlineResponse200M1Enum {
    A = 'a',
    B = 'b',
    C = 'c'
}
export enum InlineResponse200M2Enum {
    NUMBER_1 = 1,
    NUMBER_2 = 2,
    NUMBER_3 = 3
}

export interface Response {
    'enum1'?: ResponseEnum1Enum;
    'enum2'?: ResponseEnum2Enum;
}

export enum ResponseEnum1Enum {
    A = 'a',
    B = 'b',
    C = 'c'
}
export enum ResponseEnum2Enum {
    E = 'e',
    D = 'd',
    F = 'f'
}
```

**Comment by kasperpeulen** [comment]
Would be great to have this for typescript-fetch as well @ty-v1
